contracts/DomainRegistry.sol

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+/// @title Domain Registry Contract
+/// @notice This contract allows users to register and manage top-level domain names
+/// @author developed by Ivan Myasoyedov
+contract DomainRegistry {
+    /// @notice Address of the contract owner
+    address public owner;
+
+    /// @notice Fee required to register a domain
+    uint256 public registrationFee;
+
+    /// @dev Struct to store information about a domain
+    struct Domain {
+        string name;
+        address controller;
+        uint256 registeredAt;
+    }
+
+    /// @dev Array of all domains registered in the contract
+    Domain[] public domains;
```

Comment:
line 22 - 28 
It is expansive to store same data in different places
maybe you can find a way to store the data in one place 
// - 1 point